rbd-target-api.py

```diff
@@ -181,8 +181,8 @@ def _parse_target_controls(controls_json):
     controls = {}
     raw_controls = json.loads(controls_json)
     for k in settings.Settings.GATEWAY_SETTINGS:
-        if raw_controls.has_key(k):
-            value = raw_controls[k] if raw_controls[k] else None
+        if k in raw_controls:
+            value = raw_controls.get(k)
```

Comment:
I think I remember the rationale for the odd syntax: if ```raw_controls``` contains an empty string, we want to replace it with None so that ```ceph-iscsi-config``` will clear the config override and restore it to the default [1].

 [1] https://github.com/ceph/ceph-iscsi-config/blob/master/ceph_iscsi_config/gateway.py#L109

Reply:
How about:
```
if k in raw_controls and raw_controls[k]:
```

Reply:
We need the ```None``` to indicate that this particular key should be reset to default. You can just add an ```else``` block to the ```if``` on 186 that explicitly sets ```value = None```

Reply:
Done.